src/main/java/org/reaktivity/nukleus/socks/internal/stream/client/ConnectReplyStreamProcessor.java

```diff
@@ -0,0 +1,422 @@
+        final long correlationId = begin.correlationId();
+        correlation = context.correlations.remove(correlationId);
+        if (connectRef == 0L && correlation != null)
+        {
+            context.router.setThrottle(
+                correlation.acceptSourceName(),
+                correlation.acceptReplyStreamId(),
+                this::handleAcceptReplyThrottle);
+            final long acceptReplyRef = 0; // Bi-directional reply
+            BeginFW beginToAcceptReply = context.beginRW
+                .wrap(context.writeBuffer, 0, context.writeBuffer.capacity())
+                .streamId(correlation.acceptReplyStreamId())
+                .source("socks")
+                .sourceRef(acceptReplyRef)
+                .correlationId(correlation.acceptCorrelationId())
+                .extension(e -> e.reset())
+                .build();
+            correlation.acceptReplyEndpoint()
+                .accept(
+                    beginToAcceptReply.typeId(),
+                    beginToAcceptReply.buffer(),
+                    beginToAcceptReply.offset(),
+                    beginToAcceptReply.sizeof());
+
+            streamState = this::beforeNegotiationResponse;
+            doWindow(
+                connectReplyThrottle,
+                connectReplyStreamId,
+                65535,
+                65535
+            );
+        }
+        else
+        {
+            doReset(connectReplyThrottle, connectReplyStreamId);
+        }
+    }
+
+    private void handleNegotiationResponse(DataFW data)
+    {
+        OctetsFW payload = data.payload();
+        DirectBuffer buffer = payload.buffer();
+        int limit = payload.limit();
+        int offset = payload.offset();
+        int size = limit - offset;
+        // Fragmented writes might have already occurred
+        if (slotIndex != NO_SLOT)
+        {
+            // Append incoming data to the buffer
+            MutableDirectBuffer acceptBuffer = context.bufferPool.buffer(slotIndex, slotOffset);
+            acceptBuffer.putBytes(0, buffer, offset, size);
+            slotOffset += size;                                  // Next starting point is moved to the end of the buffer
+            buffer = context.bufferPool.buffer(slotIndex);       // Try to decode from the beginning of the buffer
+            offset = 0;                                          //
+            limit = slotOffset;                                  //
+        }
+        // one negotiation request frame is in the buffer
+        if (context.socksNegotiationResponseRO.canWrap(buffer, offset, limit))
+        {
+            // Wrap the frame and extract the incoming data
+            final SocksNegotiationResponseFW socksNegotiationResponse =
+                context.socksNegotiationResponseRO.wrap(buffer, offset, limit);
+            if (socksNegotiationResponse.version() != 0x05)
+            {
+                throw new IllegalStateException(
```

Comment:
this should trigger a reset of the stream (here and all exceptions).  Creating a new Exception and then catching it will create garbage.  We changed the model recently that any uncaught exception is a system failure and should restart the whole system.  This code does not mesh with that here as it allows a client to crash the system.

Reply:
Will change this. However, the socks flyweights are throwing exceptions when unable to decode. Also the flyweights generated (like DataFW) have checks that throw IllegalStateException. Is it safe to assume that these are "out of protocol" exceptions and the system should be notified/restarted if they occur? A malformed socks request would still potentially allow the system to be crashed. Should maybe use negative sizes or other conventions when returning values in case of errors ?

Reply:
> Also the flyweights generated (like DataFW) have checks that throw IllegalStateException.

There are 2 cases, 1) internal system error exceptions.   2) external system behavior failure.

The illegal State exception in DataFW is the first case and means some code in the reaktivity stack is broken and that should be addressed.  This should never happen in production/well-behaved nukleus code and should not be able to be caused by exterior systems.  As such throwing an exception here is perfect.

In contrast, a malformed Socks request should not cause a system crash as it can be influenced by an external client.  I'm alright filing tickets to clean this up after the PR, or fixing it as part of it.  It would be best not to create garbage and throw an exception from the flyweight and then catch it from the StreamFactory/Processor, but it would be an okay starting point if there is no appropriate error value OR no method that performs the check (`canwrap()` maybe?)